### DISCUSSION TITLE
How do I configure the provider and resources to use /:comicId/chapters/:id ?

### DISCUSSION THREAD

**Discussion by DoCongSon** [refinedev/refine#6175] [Q&A] [answered]
How do I configure the provider and resources to use /:comicId/chapters/:id ?
The result I desire should look something like:
```jsx
<Refine
  resources={[
    {
      name: 'chapters',
      list: '/comics/:comicId/chapters', // http://localhost:3000/comics/123/chapters
      show: '/comics/:comicId/chapters/:id', // http://localhost:3000/comics/123/chapters/1
      create: '/comics/:comicId/chapters/create', // http://localhost:3000/comics/123/chapters/create
      edit: '/comics/:comicId/chapters/:id/edit', // http://localhost:3000/comics/123/chapters/1/edit
      clone: '/comics/:comicId/chapters/:id/clone' // http://localhost:3000/comics/123/chapters/1/clone
    }
  ]}
/>
```

**Comment by aliemir** [comment, marked as answer]
This is easily achievable with Refine. When in some other page that doesn't have `:comicId` parameter and you're about to redirect to the `chapters` resource, you'll need to provide the `comicId` in `meta` prop of the navigation method if you're using the ones that Refine provides.

Such as, if you're using `<ListButton />`:

```ts
<ListButton resource="chapters" meta={{ comicId: "123" }} />
```

This will populate the `/:comicId/` segment in the route and navigate to the requested list action of the "chapters" resource. Once it is present in the URL params, then Refine will infer the `:comicId` from the URL and keep using it. 

If you want to have a default `:comicId` defined for your resource, you can use the `meta` of the resource definition and pass `comicId: "123"`, this will be used as a fallback when `comicId` is not present in the URL and not provided explicitly to the meta prop.

**Reply by DoCongSon** [reply]
I have used it and successfully redirected.
```jsx
<ListButton resource="chapters" meta={{ comicId: "123" }} />
```

But I cannot get userId from the Provider.
```jsx
const dataProvider: DataProvider = {
  getList: async ({ resource, pagination, sorters, filters, meta }) => {
      // I want to get the userId here.
      console.log('🚀 ~ meta:', meta) // undefined
  }
}
```
My APIs
![image](https://github.com/user-attachments/assets/8abc72c7-b5ac-47c1-9919-81371572443f)

**Reply by aliemir** [reply]
In your page, can you check by using `useParsed` hook if you can access the `comicId` value? `meta` value of the `getList` method should include all active meta values including the router params. Do you have the latest version for `@refinedev/core` and your routing package?

Please check your versions and it will be great if you can also test out the `useParsed` hook like I mentioned above, then If the issue continues we can move this to an issue 🙏

**Reply by DoCongSon** [reply]
After some time of tinkering, I found that when I used the `Inferencer` component to generate views, the `meta` was undefined. I tried removing it, and it worked fine.

Thank you for assisting me ❤️
